src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -4203,7 +4203,22 @@ LogicalResult ONNXReduceSumSquareOp::inferShapes(
 
 LogicalResult ONNXRoiAlignOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  int64_t height = output_height();
+  int64_t width  = output_width();
+
+  auto xTy = X().getType().cast<RankedTensorType>();
+  auto xShape = xTy.getShape();
+  auto biTy = batch_indices().getType().cast<RankedTensorType>();
+  auto biShape = biTy.getShape();
+
+  SmallVector<int64_t, 4> outputDims = { biShape[0], xShape[1], height, width};
```

Comment:
Be aware that your code can only handle tensor dimension with constant size. When a dimension size is only known dynamically (which is represented by -1), the relationship between the input and output (e.g. biShape[0] is the first dimension) is not represented in the code. We have other utilities to represent such relationship. Since this is your first patch, you can add the code for dynamic size later.